src/dashboard/src/components/backlog/views.py

```diff
@@ -71,6 +74,31 @@ def check_and_remove_deleted_transfers(es_client):
             elasticSearchFunctions.remove_backlog_transfer(es_client, transfer_uuid)
 
 
+def check_and_update_transfer_pending_deletion(uuid, pending_deletion):
+    """Check Storage Service to see if transfer has pending deletion and update ES if so.
+
+    :param uuid: Transfer UUID.
+    :param pending_deletion: Current pending_deletion value in transfers ES index.
+    :return: None
+    """
+    api_results = AMClient(
+        ss_api_key=get_setting("storage_service_apikey", None),
+        ss_user_name=get_setting("storage_service_user", "test"),
+        ss_url=get_setting("storage_service_url", None).rstrip("/"),
```

Comment:
I don't know if you need the explicit `None` as default here. I haven't a convincing default you can set. Though `86` probably shouldn't be `test`. And `87` will return an `AttributeError` if it really did return `None`. So you could have string type returns for all three, e.g. `""` but there might be a trickle down error to deal with. I think if blank strings as defaults hits the path on `90`:
```
    if not api_results:
        logger.warning("AIP not found in storage service: {}".format(uuid))
        return
```
Then that's probably the best approach. IDK what do you think?

Reply:
I've replaced the default values with empty strings in all three cases (the values that had been before were copied from another usage of amclient elsewhere in the codebase, and you're right that they weren't helping).

On closer look at amclient, errors result in an int status code being returned, not None (as I was trying to catch here). I've added handling for the three error codes amclient can return for this particular method, including cases where the connection can't be made to the Storage Service, and made sure that the log message will include the status code for debugging.

Reply:
> (the values that had been before were copied from another usage of amclient elsewhere in the codebase

Oof! I bet I know who that was!! (the editor's eye is keener on other's work :slightly_smiling_face: )

Reply:
AMClient's responses could be cleaner. We've a ticket for that: https://github.com/archivematica/Issues/issues/488 we'll get there.